version/version.go

```diff
@@ -5,25 +5,36 @@
 package version
 
 import (
-	"fmt"
+	_ "embed"
+	"os"
+	"strings"
 
 	version "github.com/hashicorp/go-version"
 )
 
 // The main version number that is being run at the moment.
-var Version = "1.5.0"
+var Version string
 
 // A pre-release marker for the version. If this is "" (empty string)
 // then it means that it is a final release. Otherwise, this is a pre-release
 // such as "dev" (in development), "beta", "rc1", etc.
-var Prerelease = "dev"
+var Prerelease string
 
 // SemVer is an instance of version.Version. This has the secondary
 // benefit of verifying during tests and init time that our version is a
 // proper semantic version, which should always be the case.
 var SemVer *version.Version
 
+// rawVersion is the current version as a string, as read from the VERSION
+// file. This must be a valid semantic version.
+//
+//go:embed VERSION
+var rawVersion string
+
 func init() {
+	rawVersion = strings.TrimSpace(rawVersion)
+	version.Must(version.NewVersion(rawVersion))
+	Version, Prerelease = extractPrerelease(rawVersion)
```

Comment:
I like this! But I think you don't need to parse the prerelease yourself. Can you do this?

```
parsed := version.Must(version.NewVersion(rawVersion))
Version = parsed.Core().String()
Prerelease = parsed.Prerelease()
```

Reply:
Good point! Reworked this a bit.